src/fileScopedNamespaceConverter.ts

```diff
@@ -0,0 +1,86 @@
+import CsprojReader from './csprojReader';
+import { Uri, workspace } from 'vscode';
+import * as path from 'path';
+import * as findupglob from 'find-up-glob';
+
+export class FileScopedNamespaceConverter {
+    /**
+     * If the file to be created is a C# file, 
+     * and the TargetFramework version of the current project is .NET 6.0+, 
+     * and this feature is enabled in settings, 
+     * then return the file-scoped namespace form of the template. 
+     * 
+     * Else return the original template.
+     * 
+     * @param template The content of the C# template file.
+     * @param filePath The path of the C# file that is being created. Used to locate the .csproj file and get the TargetFramework version.
+     */
+
+    public async getFileScopedNamespaceFormOfTemplateIfNecessary(template: string, filePath: string): Promise<string> {
+        const useFileScopedNamespace = 
+            workspace.getConfiguration().get<boolean>('csharpextensions.useFileScopedNamespace', false) &&
+            filePath.endsWith('.cs') &&
+            await this.targetFrameworkHigherThanOrEqualToDotNet6(filePath);
+
+        return useFileScopedNamespace ? this.getFileScopedNamespaceFormOfTemplate(template) : template;
+    }
+
+    private async targetFrameworkHigherThanOrEqualToDotNet6(filePath: string): Promise<boolean> {
+        const csprojs: string[] = await findupglob('*.csproj', { cwd: path.dirname(filePath) });
+
+        if (csprojs === null || csprojs.length < 1) {
+            return false;
+        }
+
+        const csprojFile = csprojs[0];
+        const fileContent =  await (await workspace.openTextDocument(Uri.file(csprojFile))).getText();
```

Comment:
VSCode tells me the second await does not do anything?
![image](https://user-images.githubusercontent.com/20839414/150773393-caa045be-3892-4f29-a94a-e00edb8c5254.png)

Readability is not good with lines like these. Could you split this up?